### PR TITLE
PT-4990: Highlight Active left menu item

### PR DESCRIPTION
**Pull request by BelyakovSergey**
## Description
Highlight Active left menu item
### Jira-link:
https://virtocommerce.atlassian.net/browse/PT-4990

### CODE REVIEW

File: src/VirtoCommerce.Platform.Core/Settings/ItemMainMenuState.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VirtoCommerce.Platform.Core.Settings
+{
+    public class ItemMainMenuState
```
Comment (akak1977) [inline src/VirtoCommerce.Platform.Core/Settings/ItemMainMenuState.cs:7]: Please add doccomment.

File: src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs [hunk 2]
```diff
@@ -42,6 +49,20 @@ public async Task<ActionResult> GetCurrentUserProfileAsync()
                 var userProfile = AbstractTypeFactory<UserProfile>.TryCreateInstance();
                 userProfile.Id = currentUser.Id;
                 await _settingsManager.DeepLoadSettingsAsync(userProfile);
+
+                var nameMainMenuState = PlatformConstants.Settings.UserProfile.MainMenuState.Name;
```
Comment (akak1977) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs:53]: Please add comment

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/appsettings.json [hunk 3]
```diff
@@ -205,5 +205,54 @@
                 "PatternUrl": "/images/pattern-live.svg"
             }
         ]
+    },
+    "MainMenuState": {
```
Comment (akak1977) [inline src/VirtoCommerce.Platform.Web/appsettings.json:209]: DefaultMainMenuState

File: src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs [hunk 4]
```diff
@@ -42,6 +49,20 @@ public async Task<ActionResult> GetCurrentUserProfileAsync()
...
+                if (userProfile.Settings.FirstOrDefault(x => x.Name == nameMainMenuState).Value == null)
+                {
+                    var settingMenuState = new List<ItemMainMenuState>();
+                    _configuration.GetSection("MainMenuState:items").Bind(settingMenuState);
```
Comment (akak1977) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs:57]: DefaultMainMenuState

File: src/VirtoCommerce.Platform.Core/Settings/ItemMainMenuState.cs (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Core/Settings/ItemMainMenuState.cs:7]: ItemDefaultMainMenuState

File: src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs [hunk 5]
```diff
@@ -42,6 +49,21 @@ public async Task<ActionResult> GetCurrentUserProfileAsync()
                 var userProfile = AbstractTypeFactory<UserProfile>.TryCreateInstance();
                 userProfile.Id = currentUser.Id;
                 await _settingsManager.DeepLoadSettingsAsync(userProfile);
+
+                // Main menu settings at initial boot
+                var nameMainMenuState = PlatformConstants.Settings.UserProfile.MainMenuState.Name;
+                if (userProfile.Settings.FirstOrDefault(x => x.Name == nameMainMenuState).Value == null)
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs:55]: Possible Null reference exception, please use:
userProfile.Settings.FirstOrDefault(x => x.Name == nameMainMenuState)?.Value == null

File: src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs [hunk 6]
```diff
@@ -42,6 +49,21 @@ public async Task<ActionResult> GetCurrentUserProfileAsync()
...
+                {
+                    var settingMenuState = new List<ItemDefaultMainMenuState>();
+                    _configuration.GetSection("DefaultMainMenuState:items").Bind(settingMenuState);
+                    var serializeOptions = new JsonSerializerOptions
+                    {
+                        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                    };
+                    var mainMenuState = $"{{\"items\": {JsonSerializer.Serialize(settingMenuState, serializeOptions)} }}";
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/ProfilesController.cs:63]: add new DefaultMainMenuState class for both serialization and deserialization

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**